docs/crocks/Reader.md

```diff
@@ -0,0 +1,61 @@
+# Reader
+
+`Reader a`
+
+--
+
+--
+
+```js
+--
+```
+
+`Reader` exposes the following functions on the constructor and instance:
+
+| Constructor | Instance |
+|:---|:---|
+| [`ask`](#ask), [`of`](#of), [`type`](#type) |  [`ap`](#ap), [`chain`](#chain), [`inspect`](#inspect), [`map`](#map), [`of`](#of), [`run`](#runWith), [`type`](#type) |
+
+## Constructor
+
+### ask
+
+`Reader m => a -> m a`
```

Comment:
Ask takes a function as its input. this is used a lot for chaining.
It asks for what the environment is and passes it to function you provide.
So looking below, that x is the value that is passed into `f`. `f` lifts it into a `Reader`. This is the `a` in the siggy.
So anytimne that `ask` is used it will always pass the `e` to it. In this case `5`.

But instead of numbers think of `e` being an object that holds a database connection, or some global configuration that is used by a family of functions in a flow.

```js
const { add, multiply } = require('ramda')
const { Reader, compose, chain } = require('../crocks')

const { ask } = Reader

const f = compose(
  chain(x => ask(multiply(x))),
  chain(x => ask(add(x))),
  Reader.of
)

log(
  f(10).runWith(5)
)

// logs 75
```